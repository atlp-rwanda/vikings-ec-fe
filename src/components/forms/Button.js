import React from 'react';
import propTypes from 'prop-types';

function Button({ label, onClick, className }) {
  return (
    <div>
      <button onClick={onClick} className={className} type="submit">
        {label}
      </button>
    </div>
  );
}

Button.propTypes = {
  label: propTypes.string,
  onClick: propTypes.func,
  className: propTypes.string,
};
Button.defaultProps = {
  label: '',
  onClick() {},
  className: '',
};
export default Button;
