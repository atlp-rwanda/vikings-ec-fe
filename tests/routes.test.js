import * as React from 'react';
import { expect, describe, it } from '@jest/globals';
import { waitFor } from '@testing-library/react';
import renderRoute, { simpleRender, renderPath } from './jest.setup';
import Button from '../src/components/forms/Button';
import 'setimmediate';

describe('My app', () => {
  it('renders home correctly', async () => {
    const renderer = renderRoute('/');
    await waitFor(() => {
      expect(renderer.toJSON()).toMatchSnapshot();
    });
  });
  it('renders auth correctly', async () => {
    const renderer = renderRoute('/auth');
    await waitFor(() => {
      expect(renderer.toJSON()).toMatchSnapshot();
    });
  });
  it('renders redirect-google correctly', async () => {
    const renderer = renderRoute('/redirect-google');
    await waitFor(() => {
      expect(renderer.toJSON()).toMatchSnapshot();
    });
  });
  it('renders auth correctly', async () => {
    const renderer = renderRoute('/auth/signup');
    waitFor(() => {
      expect(renderer.toJSON()).toMatchSnapshot();
    });
  });
  it('renders auth correctly', async () => {
    const renderer = renderRoute('/auth/signin');
    waitFor(() => {
      expect(renderer.toJSON()).toMatchSnapshot();
    });
  });
  it('renders button ', async () => {
    const renderer = simpleRender(<Button onClick={() => {}}>text</Button>);
    await waitFor(() => {
      expect(renderer.toJSON()).not.toBe(null);
    });
  });
  it('renders verify auth correctly', async () => {
    const renderer = renderRoute('auth/verify');
    await waitFor(() => {
      expect(renderer.toJSON()).toMatchSnapshot();
    });
  });
  it('renders resetPassword correctly', async () => {
    const renderer = renderRoute('/auth/reset-password?token="some');
    await waitFor(() => {
      expect(renderer.toJSON()).toMatchSnapshot();
    });
  });
  it('renders landing', async () => {
    const renderer = renderRoute('/auth/reset-password');
    await waitFor(() => {
      expect(renderer.toJSON()).toMatchSnapshot();
    });
  });
  it('renders forgot-password correctly', async () => {
    const renderer = renderRoute('/auth/forgot-password');
    await waitFor(() => {
      expect(renderer.toJSON()).toMatchSnapshot();
    });
  });
  it('renders edit profile page correct', async () => {
    const renderer = renderRoute('/profile/update');
    await waitFor(() => { expect(renderer.toJSON()).toMatchSnapshot(); });
  });
  it('renders verify email correctly', async () => {
    const renderer = renderRoute('/auth/verify-email/token');
    await waitFor(() => {
      expect(renderer.toJSON()).toMatchSnapshot();
    });
  });
  it('renders create product correctly', async () => {
    const renderer = renderRoute('/dashboard/products/create');
    await waitFor(() => {
      expect(renderer.toJSON()).toMatchSnapshot();
    });
  });
  it('renders dashboard', async () => {
    const renderer = renderRoute('/dashboard');
    await waitFor(() => {
      expect(renderer.toJSON()).toMatchSnapshot();
    });
  });
  it('renders dashboard', async () => {
    const renderer = renderRoute('/dashboard/orders');
    await waitFor(() => {
      expect(renderer.toJSON()).toMatchSnapshot();
    });
  });
  it('renders dashboard', async () => {
    const renderer = renderRoute('/dashboard/sales');
    await waitFor(() => {
      expect(renderer.toJSON()).toMatchSnapshot();
    });
  });
  it('renders update product correctly', async () => {
    const renderer = renderRoute('/dashboard/products/b5e75a01-5e67-44ad-91bd-f36ab3564a48');
    await waitFor(() => {
      expect(renderer.toJSON()).toMatchSnapshot();
    });
  });
  it('rends dashboard Correctly', async () => {
    const renderer = renderRoute('/seller-products');
    await waitFor(() => {
      expect(renderer.toJSON()).toMatchSnapshot();
    });
  });
  it('renders Users data correctly', async () => {
    const renderer = renderRoute('/dashboard/users');
    await waitFor(() => {
      expect(renderer.toJSON()).toMatchSnapshot();
    });
  });
  it('renders Users data correctly', async () => {
    const renderer = renderRoute('/notifications');
    await waitFor(() => {
      expect(renderer.toJSON()).toMatchSnapshot();
    });
  });
});

