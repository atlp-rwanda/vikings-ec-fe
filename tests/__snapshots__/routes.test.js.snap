// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My app renders auth correctly 1`] = `
<div
  className="flex flex-col h-screen bg-login bg-center"
>
  <div
    className=" bg-[#64B937] w-full justify-start py-4 hidden sm:flex "
  >
    <div
      className="ml-24 text-white"
    >
      Welcome to our online shop
    </div>
  </div>
  <div
    className="flex flex-col w-full h-full px-10 md:px-24 xl:px-60 xs:px-2"
  >
    <nav
      className="py-2.5 flex items-center justify-between w-full"
    >
      <a
        className="flex items-center"
        href="/"
      >
        <img
          alt="Vikings Logo"
          className="h-6 sm:h-9"
          src=""
        />
        <span
          className="self-center text-xl font-semibold whitespace-nowrap"
        >
          Vikings
        </span>
      </a>
      <button
        aria-controls="navbar-default"
        aria-expanded="false"
        className="inline-flex items-center p-2 ml-3 text-sm text-gray-500 rounded-lg md:hidden hover:bg-gray-100 focus:outline-none focus:ring-2 focus:ring-gray-200"
        data-collapse-toggle="navbar-default"
        type="button"
      >
        <span
          className="sr-only"
        >
          Open main menu
        </span>
        <svg
          aria-hidden="true"
          className="w-6 h-6"
          fill="currentColor"
          viewBox="0 0 20 20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clipRule="evenodd"
            d="M3 5a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1zM3 10a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1zM3 15a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1z"
            fillRule="evenodd"
          />
        </svg>
      </button>
      <ul
        className="md:flex hidden flex-1 flex-col p-4 mt-4 justify-between max-w-[54rem] rounded-lg bg-inherit bg-gray-50 md:flex-row md:space-x-8 md:mt-0 md:text-sm md:font-medium md:border-0 md:bg-white"
      >
        <li>
          <a
            className="py-2 pl-3 pr-4 text-gray-700 text-[20px] rounded hover:bg-gray-100 md:hover:bg-transparent md:border-0 md:hover:text-[#338E03] md:p-0 "
            href="/auth/Home"
          >
            Home
          </a>
        </li>
        <li>
          <a
            className="py-2 pl-3 pr-4 text-gray-700 text-[20px] rounded hover:bg-gray-100 md:hover:bg-transparent md:border-0 md:hover:text-[#338E03] md:p-0 "
            href="/auth/About"
          >
            About
          </a>
        </li>
        <li>
          <a
            className="py-2 pl-3 pr-4 text-gray-700 text-[20px] rounded hover:bg-gray-100 md:hover:bg-transparent md:border-0 md:hover:text-[#338E03] md:p-0 "
            href="/auth/Pages"
          >
            Pages
          </a>
        </li>
        <li>
          <a
            className="py-2 pl-3 pr-4 text-gray-700 text-[20px] rounded hover:bg-gray-100 md:hover:bg-transparent md:border-0 md:hover:text-[#338E03] md:p-0 "
            href="/auth/Contact"
          >
            Contact
          </a>
        </li>
        <li>
          <a
            className="py-2 pl-3 pr-4 text-gray-700 text-[20px] rounded hover:bg-gray-100 md:hover:bg-transparent md:border-0 md:hover:text-[#338E03] md:p-0 "
            href="/auth/Signin"
          >
            Signin
          </a>
        </li>
        <li>
          <a
            className="py-2 pl-3 pr-4 text-gray-700 text-[20px] rounded hover:bg-gray-100 md:hover:bg-transparent md:border-0 md:hover:text-[#338E03] md:p-0 "
            href="/auth/Signup"
          >
            Signup
          </a>
        </li>
      </ul>
    </nav>
  </div>
</div>
`;

exports[`My app renders auth correctly 2`] = `
<div
  className="flex flex-col h-screen bg-login bg-center"
>
  <div
    className=" bg-[#64B937] w-full justify-start py-4 hidden sm:flex "
  >
    <div
      className="ml-24 text-white"
    >
      Welcome to our online shop
    </div>
  </div>
  <div
    className="flex flex-col w-full h-full px-10 md:px-24 xl:px-60 xs:px-2"
  >
    <nav
      className="py-2.5 flex items-center justify-between w-full"
    >
      <a
        className="flex items-center"
        href="/"
      >
        <img
          alt="Vikings Logo"
          className="h-6 sm:h-9"
          src=""
        />
        <span
          className="self-center text-xl font-semibold whitespace-nowrap"
        >
          Vikings
        </span>
      </a>
      <button
        aria-controls="navbar-default"
        aria-expanded="false"
        className="inline-flex items-center p-2 ml-3 text-sm text-gray-500 rounded-lg md:hidden hover:bg-gray-100 focus:outline-none focus:ring-2 focus:ring-gray-200"
        data-collapse-toggle="navbar-default"
        type="button"
      >
        <span
          className="sr-only"
        >
          Open main menu
        </span>
        <svg
          aria-hidden="true"
          className="w-6 h-6"
          fill="currentColor"
          viewBox="0 0 20 20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clipRule="evenodd"
            d="M3 5a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1zM3 10a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1zM3 15a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1z"
            fillRule="evenodd"
          />
        </svg>
      </button>
      <ul
        className="md:flex hidden flex-1 flex-col p-4 mt-4 justify-between max-w-[54rem] rounded-lg bg-inherit bg-gray-50 md:flex-row md:space-x-8 md:mt-0 md:text-sm md:font-medium md:border-0 md:bg-white"
      >
        <li>
          <a
            className="py-2 pl-3 pr-4 text-gray-700 text-[20px] rounded hover:bg-gray-100 md:hover:bg-transparent md:border-0 md:hover:text-[#338E03] md:p-0 "
            href="/auth/Home"
          >
            Home
          </a>
        </li>
        <li>
          <a
            className="py-2 pl-3 pr-4 text-gray-700 text-[20px] rounded hover:bg-gray-100 md:hover:bg-transparent md:border-0 md:hover:text-[#338E03] md:p-0 "
            href="/auth/About"
          >
            About
          </a>
        </li>
        <li>
          <a
            className="py-2 pl-3 pr-4 text-gray-700 text-[20px] rounded hover:bg-gray-100 md:hover:bg-transparent md:border-0 md:hover:text-[#338E03] md:p-0 "
            href="/auth/Pages"
          >
            Pages
          </a>
        </li>
        <li>
          <a
            className="py-2 pl-3 pr-4 text-gray-700 text-[20px] rounded hover:bg-gray-100 md:hover:bg-transparent md:border-0 md:hover:text-[#338E03] md:p-0 "
            href="/auth/Contact"
          >
            Contact
          </a>
        </li>
        <li>
          <a
            className="py-2 pl-3 pr-4 text-gray-700 text-[20px] rounded hover:bg-gray-100 md:hover:bg-transparent md:border-0 md:hover:text-[#338E03] md:p-0 "
            href="/auth/Signin"
          >
            Signin
          </a>
        </li>
        <li>
          <a
            className="py-2 pl-3 pr-4 text-gray-700 text-[20px] rounded hover:bg-gray-100 md:hover:bg-transparent md:border-0 md:hover:text-[#338E03] md:p-0 "
            href="/auth/Signup"
          >
            Signup
          </a>
        </li>
      </ul>
    </nav>
    <div
      className="flex flex-col w-full md:flex-row md:h-full md:justify-between justify-center items-center xs:flex-grow"
    >
      <img
        alt=""
        className="absolute inset-x-0 bottom-0 pointer-events-none"
        src=""
      />
      <div
        className="flex flex-col xs:items-center xs:w-full xs:px-6 "
      >
        <h1
          className="text-3xl lg:text-[50px] mb-[3rem] "
        >
          Sign Up
        </h1>
        <p
          className="text-xl  xs:w-full "
        >
          already a user
          <br />
          <a
            className="text-[#338E03] pt-2"
            href="/auth/signin"
          >
            sign in!
          </a>
        </p>
      </div>
      <div
        className="hidden sm:flex "
      >
        <img
          alt=""
          src=""
        />
      </div>
      <div
        className="flex flex-col max-w-sm w-full "
      >
        <form
          className="my-8 px-5"
          onSubmit={[Function]}
        >
          <div
            className="flex flex-col -space-y-px rounded-md shadow-sm"
          >
            <div>
              <div
                className="h-[80px]"
              >
                <label
                  className="undefinedsr-only""
                  htmlFor="firstname"
                />
                <input
                  className=" w-full rounded-md px-2 py-3 placeholder:text-gray-400 sm:text-[12px] my-2 focus:bg-[#EAF0F7] bg-[#EAF0F7]"
                  name="firstname"
                  onBlur={[Function]}
                  onChange={[Function]}
                  placeholder="firstname"
                  type="text"
                />
              </div>
            </div>
            <div>
              <div
                className="h-[80px]"
              >
                <label
                  className="undefinedsr-only""
                  htmlFor="lastname"
                />
                <input
                  className=" w-full rounded-md px-2 py-3 placeholder:text-gray-400 sm:text-[12px] my-2 focus:bg-[#EAF0F7] bg-[#EAF0F7]"
                  name="lastname"
                  onBlur={[Function]}
                  onChange={[Function]}
                  placeholder="lastname"
                  type="text"
                />
              </div>
            </div>
            <div>
              <div
                className="h-[80px]"
              >
                <label
                  className="undefinedsr-only""
                  htmlFor="email"
                />
                <input
                  className=" w-full rounded-md px-2 py-3 placeholder:text-gray-400 sm:text-[12px] my-2 focus:bg-[#EAF0F7] bg-[#EAF0F7]"
                  name="email"
                  onBlur={[Function]}
                  onChange={[Function]}
                  placeholder="youremail@gmail.com"
                  type="text"
                />
              </div>
            </div>
            <div
              className=" relative"
            >
              <div
                className="h-[80px]"
              >
                <label
                  className="undefinedsr-only""
                  htmlFor="password"
                />
                <input
                  className=" w-full rounded-md px-2 py-3 placeholder:text-gray-400 sm:text-[12px] my-2 focus:bg-[#EAF0F7] bg-[#EAF0F7]"
                  name="password"
                  onBlur={[Function]}
                  onChange={[Function]}
                  placeholder="Password"
                  type="password"
                />
              </div>
              <div
                className=" absolute right-4 top-4"
                onClick={[Function]}
              >
                <svg
                  aria-hidden="true"
                  className="svg-inline--fa fa-eye-slash "
                  data-icon="eye-slash"
                  data-prefix="far"
                  focusable="false"
                  role="img"
                  style={{}}
                  viewBox="0 0 640 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M38.8 5.1C28.4-3.1 13.3-1.2 5.1 9.2S-1.2 34.7 9.2 42.9l592 464c10.4 8.2 25.5 6.3 33.7-4.1s6.3-25.5-4.1-33.7L525.6 386.7c39.6-40.6 66.4-86.1 79.9-118.4c3.3-7.9 3.3-16.7 0-24.6c-14.9-35.7-46.2-87.7-93-131.1C465.5 68.8 400.8 32 320 32c-68.2 0-125 26.3-169.3 60.8L38.8 5.1zm151 118.3C226 97.7 269.5 80 320 80c65.2 0 118.8 29.6 159.9 67.7C518.4 183.5 545 226 558.6 256c-12.6 28-36.6 66.8-70.9 100.9l-53.8-42.2c9.1-17.6 14.2-37.5 14.2-58.7c0-70.7-57.3-128-128-128c-32.2 0-61.7 11.9-84.2 31.5l-46.1-36.1zM394.9 284.2l-81.5-63.9c4.2-8.5 6.6-18.2 6.6-28.3c0-5.5-.7-10.9-2-16c.7 0 1.3 0 2 0c44.2 0 80 35.8 80 80c0 9.9-1.8 19.4-5.1 28.2zm9.4 130.3C378.8 425.4 350.7 432 320 432c-65.2 0-118.8-29.6-159.9-67.7C121.6 328.5 95 286 81.4 256c8.3-18.4 21.5-41.5 39.4-64.8L83.1 161.5C60.3 191.2 44 220.8 34.5 243.7c-3.3 7.9-3.3 16.7 0 24.6c14.9 35.7 46.2 87.7 93 131.1C174.5 443.2 239.2 480 320 480c47.8 0 89.9-12.9 126.2-32.5l-41.9-33zM192 256c0 70.7 57.3 128 128 128c13.3 0 26.1-2 38.2-5.8L302 334c-23.5-5.4-43.1-21.2-53.7-42.3l-56.1-44.2c-.2 2.8-.3 5.6-.3 8.5z"
                    fill="currentColor"
                    style={{}}
                  />
                </svg>
              </div>
            </div>
            <div>
              <div>
                <button
                  className=" w-full my-2 h-[45px] rounded-md bg-[#338E03] py-2 px-3 text-sm font-semibold text-white hover:bg-[#64B937] my-1"
                  onClick={[Function]}
                  type="submit"
                >
                  Signin
                   
                </button>
              </div>
            </div>
          </div>
        </form>
        <div
          className="flex items-center justify-center gap-2"
        >
          <hr
            className="flex-grow border-gray-300 border-t "
          />
          <span
            className="px-4 text-[#ACADAC]"
          >
            Or continue with
          </span>
          <hr
            className="flex-grow border-gray-300 border-t"
          />
        </div>
        <div
          className="flex items-center justify-center mt-4"
        >
          <a
            className=" bg-white p-2 shadow shadow-gray-400 rounded-full"
            href="/auth/signup/undefined/users/auth/google"
            onClick={[Function]}
            type="button"
          >
            <img
              alt="google"
              src=""
            />
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`My app renders auth correctly 3`] = `
<div
  className="flex flex-col h-screen bg-login bg-center"
>
  <div
    className=" bg-[#64B937] w-full justify-start py-4 hidden sm:flex "
  >
    <div
      className="ml-24 text-white"
    >
      Welcome to our online shop
    </div>
  </div>
  <div
    className="flex flex-col w-full h-full px-10 md:px-24 xl:px-60 xs:px-2"
  >
    <nav
      className="py-2.5 flex items-center justify-between w-full"
    >
      <a
        className="flex items-center"
        href="/"
      >
        <img
          alt="Vikings Logo"
          className="h-6 sm:h-9"
          src=""
        />
        <span
          className="self-center text-xl font-semibold whitespace-nowrap"
        >
          Vikings
        </span>
      </a>
      <button
        aria-controls="navbar-default"
        aria-expanded="false"
        className="inline-flex items-center p-2 ml-3 text-sm text-gray-500 rounded-lg md:hidden hover:bg-gray-100 focus:outline-none focus:ring-2 focus:ring-gray-200"
        data-collapse-toggle="navbar-default"
        type="button"
      >
        <span
          className="sr-only"
        >
          Open main menu
        </span>
        <svg
          aria-hidden="true"
          className="w-6 h-6"
          fill="currentColor"
          viewBox="0 0 20 20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clipRule="evenodd"
            d="M3 5a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1zM3 10a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1zM3 15a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1z"
            fillRule="evenodd"
          />
        </svg>
      </button>
      <ul
        className="md:flex hidden flex-1 flex-col p-4 mt-4 justify-between max-w-[54rem] rounded-lg bg-inherit bg-gray-50 md:flex-row md:space-x-8 md:mt-0 md:text-sm md:font-medium md:border-0 md:bg-white"
      >
        <li>
          <a
            className="py-2 pl-3 pr-4 text-gray-700 text-[20px] rounded hover:bg-gray-100 md:hover:bg-transparent md:border-0 md:hover:text-[#338E03] md:p-0 "
            href="/auth/Home"
          >
            Home
          </a>
        </li>
        <li>
          <a
            className="py-2 pl-3 pr-4 text-gray-700 text-[20px] rounded hover:bg-gray-100 md:hover:bg-transparent md:border-0 md:hover:text-[#338E03] md:p-0 "
            href="/auth/About"
          >
            About
          </a>
        </li>
        <li>
          <a
            className="py-2 pl-3 pr-4 text-gray-700 text-[20px] rounded hover:bg-gray-100 md:hover:bg-transparent md:border-0 md:hover:text-[#338E03] md:p-0 "
            href="/auth/Pages"
          >
            Pages
          </a>
        </li>
        <li>
          <a
            className="py-2 pl-3 pr-4 text-gray-700 text-[20px] rounded hover:bg-gray-100 md:hover:bg-transparent md:border-0 md:hover:text-[#338E03] md:p-0 "
            href="/auth/Contact"
          >
            Contact
          </a>
        </li>
        <li>
          <a
            className="py-2 pl-3 pr-4 text-gray-700 text-[20px] rounded hover:bg-gray-100 md:hover:bg-transparent md:border-0 md:hover:text-[#338E03] md:p-0 "
            href="/auth/Signin"
          >
            Signin
          </a>
        </li>
        <li>
          <a
            className="py-2 pl-3 pr-4 text-gray-700 text-[20px] rounded hover:bg-gray-100 md:hover:bg-transparent md:border-0 md:hover:text-[#338E03] md:p-0 "
            href="/auth/Signup"
          >
            Signup
          </a>
        </li>
      </ul>
    </nav>
    <div
      className="flex flex-col md:flex-row md:h-full lg:justify-between justify-center items-center xs:flex-grow bg-login"
    >
      <img
        alt=""
        className="absolute inset-x-0 bottom-0 pointer-events-none"
        src=""
      />
      <div
        className="flex flex-col xs:items-center xs:w-full xs:px-6 "
      >
        <h1
          className="text-3xl lg:text-[50px] mb-[3rem] "
        >
          Sign In
        </h1>
        <p
          className="text-xl  xs:w-full "
        >
          if you don’t have an account you can
          <br />
          <a
            className="text-[#338E03] pt-2"
            href="/auth/signup"
          >
            Register here!
          </a>
        </p>
      </div>
      <div
        className="hidden sm:flex "
      >
        <img
          alt=""
          src=""
        />
      </div>
      <div
        className="flex flex-col max-w-sm w-full "
      >
        <form
          className="my-8 px-5"
          onSubmit={[Function]}
        >
          <div
            className="flex flex-col -space-y-px rounded-md shadow-sm"
          >
            <div>
              <div
                className="h-[80px]"
              >
                <label
                  className="undefinedsr-only""
                  htmlFor="email"
                />
                <input
                  className=" w-full rounded-md px-2 py-3 placeholder:text-gray-400 sm:text-[12px] my-2 focus:bg-[#EAF0F7] bg-[#EAF0F7]"
                  name="email"
                  onBlur={[Function]}
                  onChange={[Function]}
                  placeholder="youremail@gmail.com"
                  type="text"
                />
              </div>
            </div>
            <div
              className=" relative"
            >
              <div
                className="h-[80px]"
              >
                <label
                  className="undefinedsr-only""
                  htmlFor="password"
                />
                <input
                  className=" w-full rounded-md px-2 py-3 placeholder:text-gray-400 sm:text-[12px] my-2 focus:bg-[#EAF0F7] bg-[#EAF0F7]"
                  name="password"
                  onBlur={[Function]}
                  onChange={[Function]}
                  placeholder="Password"
                  type="password"
                />
              </div>
              <div
                className=" absolute right-4 top-4"
                onClick={[Function]}
              >
                <svg
                  aria-hidden="true"
                  className="svg-inline--fa fa-eye-slash "
                  data-icon="eye-slash"
                  data-prefix="far"
                  focusable="false"
                  role="img"
                  style={{}}
                  viewBox="0 0 640 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M38.8 5.1C28.4-3.1 13.3-1.2 5.1 9.2S-1.2 34.7 9.2 42.9l592 464c10.4 8.2 25.5 6.3 33.7-4.1s6.3-25.5-4.1-33.7L525.6 386.7c39.6-40.6 66.4-86.1 79.9-118.4c3.3-7.9 3.3-16.7 0-24.6c-14.9-35.7-46.2-87.7-93-131.1C465.5 68.8 400.8 32 320 32c-68.2 0-125 26.3-169.3 60.8L38.8 5.1zm151 118.3C226 97.7 269.5 80 320 80c65.2 0 118.8 29.6 159.9 67.7C518.4 183.5 545 226 558.6 256c-12.6 28-36.6 66.8-70.9 100.9l-53.8-42.2c9.1-17.6 14.2-37.5 14.2-58.7c0-70.7-57.3-128-128-128c-32.2 0-61.7 11.9-84.2 31.5l-46.1-36.1zM394.9 284.2l-81.5-63.9c4.2-8.5 6.6-18.2 6.6-28.3c0-5.5-.7-10.9-2-16c.7 0 1.3 0 2 0c44.2 0 80 35.8 80 80c0 9.9-1.8 19.4-5.1 28.2zm9.4 130.3C378.8 425.4 350.7 432 320 432c-65.2 0-118.8-29.6-159.9-67.7C121.6 328.5 95 286 81.4 256c8.3-18.4 21.5-41.5 39.4-64.8L83.1 161.5C60.3 191.2 44 220.8 34.5 243.7c-3.3 7.9-3.3 16.7 0 24.6c14.9 35.7 46.2 87.7 93 131.1C174.5 443.2 239.2 480 320 480c47.8 0 89.9-12.9 126.2-32.5l-41.9-33zM192 256c0 70.7 57.3 128 128 128c13.3 0 26.1-2 38.2-5.8L302 334c-23.5-5.4-43.1-21.2-53.7-42.3l-56.1-44.2c-.2 2.8-.3 5.6-.3 8.5z"
                    fill="currentColor"
                    style={{}}
                  />
                </svg>
              </div>
            </div>
            <div>
              <a
                className=" flex justify-end text-[#338e03] text-[14px] my-2"
                href=""
              >
                Forget password ?
              </a>
            </div>
            <div>
              <div>
                <button
                  className=" w-full my-2 h-[45px] rounded-md bg-[#338E03] py-2 px-3 text-sm font-semibold text-white hover:bg-[#64B937] my-1"
                  onClick={[Function]}
                  type="submit"
                >
                  Signin
                   
                </button>
              </div>
            </div>
          </div>
        </form>
        <div
          className="flex items-center justify-center gap-2"
        >
          <hr
            className="flex-grow border-gray-300 border-t "
          />
          <span
            className="px-4 text-[#ACADAC]"
          >
            Or continue with
          </span>
          <hr
            className="flex-grow border-gray-300 border-t"
          />
        </div>
        <div
          className="flex items-center justify-center mt-4"
        >
          <a
            className=" bg-white p-2 shadow shadow-gray-400 rounded-full"
            href="/auth/signin/undefined/users/auth/google"
            onClick={[Function]}
            type="button"
          >
            <img
              alt="google"
              src=""
            />
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`My app renders auth correctly 4`] = `
<div>
  <h1>
    Home Page
  </h1>
  <h2
    className="underline"
  >
    welcome
  </h2>
  <a
    href="/auth/signin"
    onClick={[Function]}
  >
    Login
  </a>
  <br />
  <a
    href="/auth/signup"
    onClick={[Function]}
  >
    Signup
  </a>
  <br />
  <a
    href="https://www.chromatic.com/builds?appId=64218d727e570b64b94415f8"
  >
    Go to storybook
  </a>
  <h1>
    "no user found"
  </h1>
</div>
`;

exports[`My app renders redirect-google correctly 1`] = `
<div
  className="flex flex-col gap-2 items-center justify-center h-screen w-screen"
>
  <h4>
    please wait...
  </h4>
  <div
    role="status"
  >
    <svg
      aria-hidden="true"
      className="w-8 h-8 mr-2 text-gray-200 animate-spin dark:text-gray-600 fill-blue-600"
      fill="none"
      viewBox="0 0 100 101"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M100 50.5908C100 78.2051 77.6142 100.591 50 100.591C22.3858 100.591 0 78.2051 0 50.5908C0 22.9766 22.3858 0.59082 50 0.59082C77.6142 0.59082 100 22.9766 100 50.5908ZM9.08144 50.5908C9.08144 73.1895 27.4013 91.5094 50 91.5094C72.5987 91.5094 90.9186 73.1895 90.9186 50.5908C90.9186 27.9921 72.5987 9.67226 50 9.67226C27.4013 9.67226 9.08144 27.9921 9.08144 50.5908Z"
        fill="currentColor"
      />
      <path
        d="M93.9676 39.0409C96.393 38.4038 97.8624 35.9116 97.0079 33.5539C95.2932 28.8227 92.871 24.3692 89.8167 20.348C85.8452 15.1192 80.8826 10.7238 75.2124 7.41289C69.5422 4.10194 63.2754 1.94025 56.7698 1.05124C51.7666 0.367541 46.6976 0.446843 41.7345 1.27873C39.2613 1.69328 37.813 4.19778 38.4501 6.62326C39.0873 9.04874 41.5694 10.4717 44.0505 10.1071C47.8511 9.54855 51.7191 9.52689 55.5402 10.0491C60.8642 10.7766 65.9928 12.5457 70.6331 15.2552C75.2735 17.9648 79.3347 21.5619 82.5849 25.841C84.9175 28.9121 86.7997 32.2913 88.1811 35.8758C89.083 38.2158 91.5421 39.6781 93.9676 39.0409Z"
        fill="currentFill"
      />
    </svg>
    <span
      className="sr-only"
    >
      Loading...
    </span>
  </div>
</div>
`;
